### PR TITLE
[PPML] BigDL Key Management Server

### PR DESCRIPTION
**Pull request by Uxito-Ada**
## Description

A reference implementation for easy-to-deploy KMS

**Original most secure design:**
image

**Fault-tolerant and simplified design:**
image

### 1. Why the change?

as above

### 2. User API changes

KMS APIs are same to bigdl-kms

### 3. Summary of the change 

Easy Key Management Server

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [x] Application test
- [ ] Document test
- [ ] ...

### 5. New dependency
sqlite-jdbc

### CODE REVIEW

File: ppml/services/bigdl-kms/kubernetes/README.md [hunk 1]
```diff
@@ -61,8 +61,10 @@ You can communicate with BigDL KMS using client [BigDLKeyManagementService](http
```
Comment (qiyuangong) [inline ppml/services/bigdl-kms/kubernetes/README.md:61]: This PR should be merged after 2.3.0 release.

File: ppml/services/easy-kms/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+RUN mkdir -p /ppml/keys && \
+    mkdir -p /ppml/password && \
+    mkdir -p /ppml/data && \
+    cd /ppml && \
+    chmod a+x /ppml/start-service.sh && \
+    gramine-argv-serializer bash -c '$sgx_command' > /ppml/secured_argvs && \
+    apt-get install sqlite3
+COPY --from=java /opt/jdk  /opt/jdk8
+# ppml
+COPY --from=intelanalytics/bigdl-ppml-trusted-bigdata-gramine-reference-8g:2.3.0-SNAPSHOT ${BIGDL_HOME}/jars/bigdl-ppml-spark_${SPARK_VERSION}-${BIGDL_VERSION}.jar /ppml/jars
```
Comment (qiyuangong) [inline ppml/services/easy-kms/docker/Dockerfile:47]: Do we need to binding with Gramine?

Comment (Uxito-Ada) [reply ppml/services/easy-kms/docker/Dockerfile:47]: Gramine is one reference implementation of KMS in SGX.
For TDX user, set `SGX_ENABLED` to false and therefore gramine will not be triggered.
For Occlum user, he can follow our implementation to develop his own image.

**Comment by Uxito-Ada** [comment]
Easy KMS user guide: https://github.com/intel-analytics/BigDL/blob/f6b065012c9e83ab944c6f31788c8831d96a5e96/ppml/services/easy-kms/kubernetes/README.md#deploy-easy-kms-key-management-service-on-kubernetes

File: ppml/services/easy-kms/kubernetes/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Deploy Easy KMS (Key Management Service) on Kubernetes
```
Comment (Uxito-Ada) [inline ppml/services/easy-kms/kubernetes/README.md:1]: Easy KMS user guide here

File: ppml/services/easy-kms/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,49 @@
+ARG BIGDL_VERSION=2.3.0-SNAPSHOT
```
Comment (glorysdj) [inline ppml/services/easy-kms/docker/Dockerfile:1]: rename to bigdl-kms

Comment (glorysdj) [inline ppml/services/easy-kms/docker/Dockerfile:1]: organize the dockerfiles into base and ref folders
one readme for bigdl-kms
two bigdl-kms-depolyment for both SGX and TDX
rename start-service to bigdl-kms-entrypoint.sh

**Review by xiangyuT — APPROVED** [review]
LGTM

**Pull request merged**